pkg/entry/mongo/document.go

```diff
@@ -158,10 +173,15 @@ func (d *LogDocument) Populate(ctx context.Context, ts time.Time, record map[int
 		logger.Debug("Key not found, use stdout", map[string]interface{}{
 			"error": err,
 		})
-
 		d.Stream = "stdout"
 	}
 
+	logPrefixPattern := "[" + logPrefix + "]"
+	if logPrefix != "" && strings.HasPrefix(logContent, logPrefixPattern) {
+		d.Stream = logPrefix + "_" + d.Stream
+		d.Log = strings.Replace(d.Log, logPrefixPattern, "", 1)
+	}
+
```

Comment:
What about parsing the prefix using following regex?

```regex
^\[(?P<logPrefix>[^\]]*)\]
```

So `logprefix` can be detected dynamically.
PS: Be careful about logs from apps and classical jobs, no prefix should be parsed.

Reply:
Usage of Regexp done, please advice